### PR TITLE
ADBDEV-4149 Support Explicit Redistribute Motion inside the SubPlans.

### PR DESCRIPTION
**Pull request by bimboterminator1**
Flow structure has a segidColIdx field, which referencing a gp_segment_id
column in plan's targetList when the Explicit Redistribute Motion is requested.
There was a problem, that _copyFlow, _equalFlow, _outFlow and _readFlow
functions did not handle the segidColIdx field. Therefore, the Flow node
could not be serialized and deserialized, or copied correctly.

The problem manifested itself when a query had UPDATE/DELETE operation, that
required Explicit Redistribute, inside the SubPlan (or InitPlan). The Explicit
Redistribute did not applied correctly inside the apply_motion_mutator function
because by that moment the value segidColIdx had been lost. The problem occured
because previously SubPlans had been mutated inside the ParallelizeSubplan
function, where the SubPlan's plan had been copied via copyObject function.
This function copies the whole plan including the Flow node, which is copied
via _copyFlow function. However the Flow node copying did not include the
copying of segidColIdx Flow field, which is used for valid performance of
Explicit Redistribute Motion.

Therefore, this patch solves the issue by adding segidColIdx to the list of
fields to copy, serialize, deserialize and compare in the _copyFlow, _outFlow,
_readFlow and _equalFlow functions respectively.

### CODE REVIEW

File: src/test/regress/expected/subselect_gp.out [hunk 1]
```diff
@@ -3112,3 +3112,201 @@ select * from r where b in (select b from s where c=10 order by c limit 2);
  1 | 2 | 3
 (1 row)
 
+-- Test that Explicit Redistribute Motion is applied properly for
```
Comment (Stolb27) [inline src/test/regress/expected/subselect_gp.out:3115]: You should modify `/src/test/regress/expected/subselect_gp_1.out` too. See that file commit history for details. You didn't face CI issues as far as ppc64le tests are muted now.

Comment (bimboterminator1) [reply src/test/regress/expected/subselect_gp.out:3115]: done

**Comment by bimboterminator1** [comment]
Have changed the patch description, tests.

**Comment by BenderArenadata** [comment]
Allure report https://allure-ee.adsw.io/launch/52537

**Review by Stolb27 — DISMISSED** [review]

**Comment by BenderArenadata** [comment]
Allure report https://allure-ee.adsw.io/launch/52647

**Comment by andr-sokolov** [comment]
@bimboterminator1 , "Residtibute" - typo

**Comment by bimboterminator1** [comment]
> "Residtibute" - typo

Fixed

**Comment by BenderArenadata** [comment]
Allure report https://allure-ee.adsw.io/launch/52668

**Comment by andr-sokolov** [comment]
I suggest to fix the _outFlow function too. And what about _readFlow?

**Comment by bimboterminator1** [comment]
Have changed the patch and PR description, the final description is in the latest commit.

**Comment by BenderArenadata** [comment]
Allure report https://allure-ee.adsw.io/launch/52699

**Review by andr-sokolov — APPROVED** [review]

**Pull request merged**